index.d.ts

```diff
@@ -0,0 +1,212 @@
+// Type definitions for neo4j-graphql-js
+// Project: https://github.com/neo4j-graphql/neo4j-graphql-js
+// Definitions by: Nopzen <https://github.com/nopzen>
+// Definitions: https://github.com/neo4j-graphql/neo4j-graphql-js.git
+// TypeScript Version: 3.8
+
+/* eslint no-unused-vars: "off" */
+
+declare module 'neo4j-graphql-js' {
+    import { Driver } from 'neo4j-driver';
+    import { GraphQLSchema, GraphQLFieldResolver, GraphQLResolveInfo, ExecutionResult, DocumentNode } from 'graphql';
+    import { IResolvers } from 'graphql-tools';
+
+    /**
+     * makeAugmentedSchema
+     * @description Wraps {@link https://www.apollographql.com/docs/apollo-server/api/apollo-server/#makeExecutableSchema|makeExecutableSchema} to create a GraphQL schema from GraphQL type definitions (SDL). Will generate Query and Mutation types for the provided type definitions and attach `neo4jgraphql` as the resolver for these queries and mutations. Either a schema or typeDefs must be provided. `resolvers` can optionally be implemented to override any of the generated Query/Mutation fields. Additional options are passed through to `makeExecutableSchema`.
+     *
+     * @param {makeAugmentedSchemaOptions} options
+     */
+    export function makeAugmentedSchema(options: makeAugmentedSchemaOptions): GraphQLSchema;
+
+    /**
+     * neo4jgraphql
+     * @description This function's signature matches that of {@link https://graphql.org/learn/execution/#root-fields-resolvers|GraphQL resolver functions}. and thus the parameters match the parameters passed into resolve by GraphQL implementations like graphql-js.
+     *
+     * It can be called within a resolver to generate a Cypher query and handle the database call to Neo4j to completely resolve the GraphQL request. Alternatively, use `cypherQuery` or `cypherMutation` within a resolver to only generate the Cypher query and handle the database call yourself.
+     * @param {object} object                   The previous object being resolved. Rarely used for a field on the root Query type.
+     * @param {RequestArguments} args           The arguments provided to the field in the GraphQL query.
+     * @param {Neo4jContext} context            Value provided to every resolver and hold contextual information about the request, such as the currently logged in user, or access to a database. neo4j-graphql-js assumes a neo4j-javascript-driver instance exists in this object, under the key driver.
+     * @param {GraphQLResolveInfo} resolveInfo  Holds field-specific information relevant to the current query as well as the GraphQL schema.
+     * @param {boolean} debug                   Specifies whether to log the generated Cypher queries for each GraphQL request. Logging is enabled by default.
+     */
+    export function neo4jgraphql(
```

Comment:
can we make neo4jgraphql a generic type, so that we get a result correctly typed ?

Would like to do something like this:
```ts
      const result = await neo4jgraphql<User>(page, params, context, resolveInfo);
     // type result = User & { FRAGMENT_TYPE: string }
```